---
title: "Mass spectrometry based metabolomics/exposomics data analysis"
subtitle: "From known to unknown"
author: "Miao Yu"
date: "`r Sys.Date()`"
output:
  xaringan::moon_reader:
    css: ["default", "my-theme.css", "middlebury"]
    lib_dir: libs
    nature:
      highlightStyle: github
      highlightLines: true
      countIncrementalSlides: false
      ratio: '16:9'
---

```{r setup, include=FALSE}
options(htmltools.dir.version = FALSE)
knitr::opts_chunk$set(fig.align='center',echo = F, cache=T,message=FALSE,warning=FALSE,comment=NA)
```

## Self-introduction

### Miao Yu

### Education + Training

- 2006 - 2010 B.S. Shandong University, Jinan, Shandong, China

- 2010 - 2016 Ph.D. University of Chinese Academy of Sciences, Beijing, China

- 2016 - 2018 PostDoc University of Waterloo, Waterloo, ON, Canada

- 2018 - now PostDoc Icahn School of Medicine at Mount Sinai, New york, NY, USA

---

## Typical workflow for HRMS metabolomics

- Sample Collection and Pretreatment

- Instrumental Analysis

- Data Analysis

---

## Sample Collection and Pretreatment

### Quenching

    - Instantaneous inactivation of metabolism
    - rapid low/high temprature
    - Organic solvent

### Storage

    - -80°C or -20°C
    - Dry ice

---

### Sample Collection 

```{r echo=FALSE, out.width= '90%'}
knitr::include_graphics('https://raw.githubusercontent.com/yufree/yufree.cn/master/static/images/TOC.png')
```

.half[
```{r refaca,results = 'asis'}
doi <- c('10.1016/j.aca.2020.08.050')
cat(unlist(rcrossref::cr_cn(dois = doi, format = "text", style = 'apa')))
```
]

---

### Sample Pretreatment

- From samples to injection vials

- More **interesting** compounds, less **unrelated** interference (matrix effects)

- Remove lipid

  - Gel Permeation Chromatograph(GPC)
  - Florisil
  - Alumina
  - Silica gel
  
- Protein denaturation
  
  - Alcohols
  - Strong acid/base

---

### Sample Pretreatment

- one-step extraction/clean up by accelerated solvent extraction (ASE)

```{r echo=FALSE, out.width= '50%'}
knitr::include_graphics('https://yufree.github.io/presentation/figure/ASE.png')
```

.half[
```{r reftal,results = 'asis'}
doi <- c('10.1016/j.talanta.2016.11.046')
cat(unlist(rcrossref::cr_cn(dois = doi, format = "text", style = 'apa')))
```
]

---

## Instrumental Analysis

### Run order effects

- `getpqsi` function in **enviGCMS** package

```{r echo=FALSE, out.width= '50%'}
knitr::include_graphics('https://yufree.github.io/presentation/figure/pooledQC.png')
```

---

## Instrumental Analysis

### Batch effects

```{r echo=FALSE, out.width= '70%'}
knitr::include_graphics('https://www.biorxiv.org/content/biorxiv/early/2019/12/17/2019.12.16.878637/F7.medium.gif')
```

.half[
```{r refpreprint,results = 'asis'}
doi <- c('10.1101/2019.12.16.878637')
cat(unlist(rcrossref::cr_cn(dois = doi, format = "text", style = 'apa')))
```
]

---

## Data Analysis

> Features -> Compounds -> Relationship among compounds

- Compounds/metabolites without standards

```{r echo=FALSE, out.width= '70%'}
knitr::include_graphics('https://yufree.github.io/presentation/figure/TOC2.png')
```

.half[
```{r refmeopbde,results = 'asis'}
doi <- c('10.1016/j.talanta.2016.01.047')
cat(unlist(rcrossref::cr_cn(dois = doi, format = "text", style = 'apa')))
```
]

---

### Gap between features and compounds

```{r echo=FALSE, out.width= '70%'}
knitr::include_graphics('https://yufree.github.io/presentation/figure/peakcom.png')
```

---

### How many real compounds among features?

```{r echo=FALSE}
knitr::include_graphics('https://yufree.github.io/presentation/figure/gapfc.jpg')
```

.half[
```{r ref1,results = 'asis'}
doi <- c('10.1021/acs.analchem.7b02380')
cat(unlist(rcrossref::cr_cn(dois = doi, format = "text", style = 'apa')))
```
]

---
### GlobalStd Algorithm

```{r echo=FALSE, out.width= '70%'}
knitr::include_graphics('https://yufree.github.io/presentation/figure/GlobalStd.png')
```

.half[
```{r ref2,results = 'asis'}
doi <- c('10.1016/j.aca.2018.10.062')
cat(unlist(rcrossref::cr_cn(dois = doi, format = "text", style = 'apa')))
```
]

---
### GlobalStd Algorithm Step 1

- Retention time cluster analysis

```{r echo=FALSE,out.width = "500px"}
library(rmwf)
data(mzrt)
mzrtm <- enviGCMS::getdoe(mzrt)
gm <- mzrtm$groupmean
gr <- mzrtm$grouprsd
# filter data
srm <- grepl('NIST',colnames(gm))
blk <- grepl('Matrix',colnames(gm))
indexmean <- apply(gm,1,function(x) all(x[srm]>= 3*x[blk]))
mzrtfilter <- enviGCMS::getfilter(mzrt,rowindex = indexmean)
mzrtnist <- enviGCMS::getfilter(mzrt,rowindex = indexmean,colindex = grepl('NIST',colnames(mzrt$data)))
pmd <- pmd::getpaired(mzrtnist)
pmd::plotrtg(pmd)
```

---
### GlobalStd Algorithm Step 2

- High frequency PMD analysis across RT clusters

```{r echo=FALSE,out.width = "500px"}
pmd::plotpaired(pmd)
```

---

### GlobalStd Algorithm Step 2

- High frequency PMD analysis across RT clusters - example

- Based on data itself, those adducts/multiply charged ions/neutral loss/isotopologues can be unknown

```{r echo=FALSE,out.width = "500px"}
index <- pmd$paired$diff2 == 21.98
pmd::plotpaired(pmd,index)
```

---
### GlobalStd Algorithm Step 3

- Independent peaks selection

```{r echo=FALSE,out.width = "500px"}
std <- pmd::getstd(pmd)
pmd::plotstd(std)
```

---
### GlobalStd Algorithm Step 3

- Independent peaks selection - example

```{r echo=FALSE,out.width = "500px"}
par(mfrow = c(2,2))
pmd::plotstdrt(std,rtcluster = 47,main = 'Retention time group 47')
pmd::plotstdrt(std,rtcluster = 44,main = 'Retention time group 44')
pmd::plotstdrt(std,rtcluster = 103,main = 'Retention time group 103')
pmd::plotstdrt(std,rtcluster = 1,main = 'Retention time group 1')
```

---
### GlobalStd Algorithm Step 3

- ~14.3% peaks can capture similar variances of all peaks
- For CAMERA/RAMclust, peaks with highest intensity from pcgroup were selected as independent peaks

```{r echo=FALSE,out.width = "600px"}
knitr::include_graphics('https://yufree.github.io/presentation/figure/software.png')
```

---

### PMDDA workflow

```{r echo=FALSE,out.width = "600px"}
knitr::include_graphics('https://yufree.github.io/presentation/figure/PMDDAraw.png')
```

.half[
```{r refpreprint2,results = 'asis'}
doi <- c('10.33774/chemrxiv-2021-fxw2c-v2')
cat(unlist(rcrossref::cr_cn(dois = doi, format = "text", style = 'apa')))
```
]

---

### PMDDA workflow

```{r echo=FALSE,out.width = "500px"}
knitr::include_graphics('https://yufree.github.io/presentation/figure/pmddacomp.png')
```

---

### Reactomics

> Features -> Compounds -> Relationship among compounds

- You ACTUALLY don't need people (compounds) name to know their relationship

```{r echo=FALSE, out.width= '50%'}
knitr::include_graphics('https://upload.wikimedia.org/wikipedia/commons/7/7d/A_Sunday_on_La_Grande_Jatte%2C_Georges_Seurat%2C_1884.jpg')
```

From [Wikipedia Commons](https://commons.wikimedia.org/wiki/File:A_Sunday_on_La_Grande_Jatte,_Georges_Seurat,_1884.jpg):A Sunday on La Grande Jatte, Georges Seurat

.half[
```{r refcc,results = 'asis'}
doi <- c('10.1038/s42004-020-00403-z')
cat(unlist(rcrossref::cr_cn(dois = doi, format = "text", style = 'apa')))
```
]

---

### Reactomics

> Sample -> Peaks -> ~~Compounds~~ -> Relationship among compounds

- Mass spectrum could directly measure relationship (reactions)

```{r echo=FALSE, out.width= '90%'}
knitr::include_graphics('https://yufree.github.io/presentation/figure/srda.png')
```

---
### Why Reactions?

```{r echo=FALSE}
knitr::include_graphics('https://yufree.github.io/presentation/figure/database.png')
```

- Unit: Gene (5) < Protein (20+2) < Metabolite (100K) < Compound (100M)

- Combination: Gene (20,000-25,000) < Protein (20,000-25,000) < Compound(???) 

- Small molecule  **combination** is a chemical reaction or paired mass distance

---

### Why PMD?

- [Nuclear Binding Energy](https://en.wikipedia.org/wiki/Nuclear_binding_energy)

$$\Delta m = Zm_{H} + Nm_{n} - M$$
- The missing mass was converted into energy ( $E=mc^2$ ) and emitted when the atom was made

- Atoms -> Compounds -> Mass distances between compounds

- **Paired Mass Distances (PMD)** is unique

- **High resolution** mass spectrometry WINs

---

### Sources of PMDs in real data

.pull-left[- Isotopologues 
  
  - $[M]^+$ $[M+1]^+$
  - 1.006 Da

- in source reaction 

  - $[M+H]^+$  $[M+Na]^+$
  - 21.982 Da
]

.pull-right[
- Homologous series

  - Lipid $-[CH_2]-$
  - 14.016 Da

- Xenobiotic metabolism

  - Phase I hydrolation
  - 15.995 Da
]  

---

### Quantitative and Qualitative analysis in Reactomics

- Reaction: $S_1 + S_2 + … + S_n = P_1 + P_2 + … + P_m \ (n>=1,m>=1)$

- PMD matrix

```{r render= 'asis'}
C1 <- c('\\(|S_1 - P_1|\\)','\\(|S_2 - P_1|\\)','...','\\(|S_n - P_1|\\)')
C2 <- c('\\(|S_1 - P_2|\\)','\\(|S_2 - P_2|\\)','...','\\(|S_n - P_2|\\)')
C3 <- c('...','...','...','...')
Cm <- c('\\(|S_1 - P_m|\\)','\\(|S_2 - P_m|\\)','...','\\(|S_n - P_m|\\)')

tb <- rbind.data.frame(C1,C2,C3,Cm)
colnames(tb) <- c('\\(S_1\\)','\\(S_2\\)','...','\\(S_n\\)')
rownames(tb) <- c('\\(P_1\\)','\\(P_2\\)','...','\\(P_m\\)')
tb <- knitr::kable(tb, align="c",format = 'html',row.names = T,escape = F)
tb <- gsub("`", "", tb) 
tb
```

- Paired mass distances for certain parent compound:

$$PMD_{S_{k}} = min(|S_k-P_1|,|S_k-P_2|,...,|S_k-P_m|) (1 \le k \le n )$$
- Paired mass distances for certain reaction:

$$PMD_R = \{PMD_{S_{1}}, PMD_{S_{2}},...,PMD_{S_{n}}\}$$

---

### Quantitative and Qualitative analysis in Reactomics (KEGG reaction database)

```{r}
example = data.frame(
  stringsAsFactors = FALSE,
                       PMD=c("2.016", "15.995", "79.966", "14.016", "0", "18.011", "162.053", "159.933", "1.032", "42.011"),
  Freq= c(1732L, 1169L, 729L, 594L, 532L, 359L, 365L, 243L, 262L, 237L),
  ExampleReactionClass=c("RC00095", "RC00046 ", "RC00002", "RC00060", "RC00302", "RC00680", "RC00049", "RC02056", "RC00006", "RC00004"),
  ExampleEnzyme = c("1.3.1.84", "1.3.99.18", "3.6.1.3", "1.5.3.1", "5.1.1.3", "3.5.2.5", "3.2.1.23", "4.2.3.42", "1.4.1.2", "2.3.1.54"))
knitr::kable(example,align = 'c',format = 'markdown')
```

---

### Quantitative and Qualitative analysis in Reactomics (HMDB compounds database)

```{r}
library(pmd)
top10 <- cbind(c(1,0,2,2,0,1,4,3,1,2),c(2,2,4,2,0,0,8,6,2,0),c(0,0,0,0,1,0,0,0,1,0))
colnames(top10) <- c('C','H','O')
rownames(top10) <- c(14.016,2.016,28.031,26.016,15.995,12,56.063,42.047,30.011,24)
knitr::kable(top10,align = 'c',format = 'markdown')
```

- Dominated by C, H and O
- Structure or reaction?

---
### Quantitative and Qualitative analysis in Reactomics (HMDB compounds database)

```{r}
ec <- c("+C2H", "+2H", "+2C4H", "+2C2H", "+O", "+C", "+4C8H", "+3C6H", "+C2HO", "+2C")
PMD3 <- c(14.016, 2.016, 28.031, 26.016, 15.995, 12, 56.063, 42.047, 30.011, 24)
PMD2 <- round(PMD3,2)
PMD1 <- round(PMD3,1)
PMD0 <- round(PMD3,0)
# freq
PMD3f <- c(4934L, 4909L, 4878L, 4229L, 4214L, 3861L, 3861L, 3771L, 3698L, 3689L)
PMD2f <- c(8003L, 7959L, 7799L, 7343L, 7731L, 7145L, 6699L, 6558L, 6761L, 6963L)
PMD1f <- c(50419L, 50467L, 50797L, 48517L, 51278L, 49335L, 36417L, 49808L, 51241L, 48099L)
PMD0f <- c(156245L, 156260L, 155410L, 154346L, 155811L, 155339L, 151894L, 153764L, 154369L, 154278L)
# accuracy
PMD3a <- c(0.98, 0.97, 0.99, 0.98, 0.99, 0.99, 0.97, 0.98, 0.95, 0.99)
PMD2a <- c(0.63, 0.62, 0.61, 0.58, 0.56, 0.56, 0.56, 0.58, 0.54, 0.53)
PMD1a <- c(0.1, 0.1, 0.09, 0.09, 0.09, 0.08, 0.1, 0.08, 0.07, 0.08)
PMD0a <- c(0.03, 0.03, 0.03, 0.03, 0.03, 0.03, 0.02, 0.03, 0.02, 0.02)

df <- cbind.data.frame(ec,PMD3,PMD3f,PMD3a)
colnames(df) <- c('Elemental Composition','PMD(digits = 3)','Frequency','Accuracy')
knitr::kable(df,align = 'c',format = 'markdown',col.names = NA)
```

---
### Quantitative and Qualitative analysis in Reactomics (HMDB compounds database)

```{r}
df2 <- cbind.data.frame(ec,PMD2,PMD2f,PMD2a)
colnames(df2) <- c('Elemental Composition','PMD(digits = 2)','Frequency','Accuracy')
knitr::kable(df2,align = 'c',format = 'markdown',col.names = NA)
```

---
### Quantitative and Qualitative analysis in Reactomics (HMDB compounds database)

```{r}
df3 <- cbind.data.frame(ec,PMD1,PMD1f,PMD1a)
colnames(df3) <- c('Elemental Composition','PMD(digits = 1)','Frequency','Accuracy')
knitr::kable(df3,align = 'c',format = 'markdown',col.names = NA)
```

---
### Quantitative and Qualitative analysis in Reactomics (HMDB compounds database)

```{r}
df4 <- cbind.data.frame(ec,PMD0,PMD0f,PMD0a)
colnames(df4) <- c('Elemental Composition','PMD(digits = 0)','Frequency','Accuracy')
knitr::kable(df4,align = 'c',format = 'markdown',col.names = NA)
```

---

### Quantitative and Qualitative analysis in Reactomics (Static v.s. dynamic)

- Static mass pairs: paired intensity ratio is stable across samples
- Dynamic mass pairs: paired intensity ratio is not stable across samples
- For example, [A,B], [C,D] and [E,F] are involved in the same PMD:

```{r}
GroupI <- c(100,50,'2:1',100,50,'2:1', 30,40, '3:4')
GroupII <- c(1000,500,'2:1',10,95,'2:19',120,160,'3:4' )
df <- rbind.data.frame(GroupI,GroupII)
colnames(df) <- c('A','B','Ins ratio','C','D','Ins ratio','E','F','Ins ratio')
knitr::kable(df,align = 'c',format = 'markdown')
```

- [A,B] and [E,F] are static mass pairs so can be used in quantitative analysis, rsd cutoff 30%
- [C,D] can be used to check dynamics of a specific reaction

---
class: inverse, center, middle

## Reactomics Application 

### Metabolites Discovery

---

## Metabolites of exogenous compound

  - Environmental pollution metabolites
  - Drug metabolites

### Xenobiotic metabolism

  - Phase I
    - Oxidation (R-H => R-OH, pmd 15.995 Da)
    - Reduction (R-C=O => R-C-OH, pmd 2.016 Da)
    
  - Phase II
    - Methylation (R-OH => R-O-C,pmd 14.016 Da)
    - Sulfation (R-OH => R-SO4, pmd 46.976 Da)
    - Acetylation (R-OH => R-O-COCH3, pmd 42.011 Da)
    - Glucuronidation (R-NH2 => R-NH-C6H9O7, pmd 192.027 Da)
    - Glycosylation (R-OH => R-O-C6H11O5, pmd 162.053 Da)

---

## Metabolites of TBBPA in Pumpkin

- Mass defect analysis to screen Brominated Compounds 

- Confirmation by synthesized standards

```{r echo=FALSE,out.width = "600px"}
knitr::include_graphics('https://yufree.github.io/presentation/figure/TBBPAmet.jpeg')
```

.half[
```{r ref3,results = 'asis'}
doi <- c('10.1021/acs.est.9b02122')
cat(unlist(rcrossref::cr_cn(dois = doi, format = "text", style = 'apa')))
```
]

---

## Metabolites of TBBPA in Pumpkin

- TBBPA Metabolites PMD network

```{r echo=FALSE,out.width = "500px"}
knitr::include_graphics("https://yufree.github.io/presentation/figure/TBBPA.png")
```


.half[
```{r ref4,results = 'asis'}
doi <- c('10.1021/acs.est.9b02122')
cat(unlist(rcrossref::cr_cn(dois = doi, format = "text", style = 'apa')))
```
]

---
class: inverse, center, middle

# Reactomics Application 

## Endogenous or Exogenous

---

## Endogenous vs Exogenous

- T3DB Endogenous (255) vs Exogenous (705)

- Use top 10 high frequency PMDs

```{r echo=FALSE,out.width = "450px"}
knitr::include_graphics("https://raw.githubusercontent.com/yufree/presentation/gh-pages/figure/t3db3.png")
```

---

## Gatekeepers for exposome

```{r echo=FALSE,out.width='90%'}
knitr::include_graphics('https://raw.githubusercontent.com/yufree/presentation/gh-pages/figure/gkpnp.png')
```

.half[
```{r refgk,results = 'asis'}
doi <- c('10.26434/chemrxiv.14781498.v1')
cat(unlist(rcrossref::cr_cn(dois = doi, format = "text", style = 'apa')))
```
]

---

## KEGG reaction network

- Metabolites of four compounds

- Average network distance: Glucose(9.74), 5-Cholestene(6.55), Caffeine(3.31), Bromophenol(1.8), TBBPA(3.88 for pumpkin)

```{r echo=FALSE,out.width = "450px"}
knitr::include_graphics("https://raw.githubusercontent.com/yufree/presentation/gh-pages/figure/kegg.png")
```


---

# Follow up studies

```{r echo=FALSE,out.width = "800px"}
knitr::include_graphics("https://raw.githubusercontent.com/yufree/presentation/gh-pages/figure/pmdif.png")
```

---

class: inverse, center, middle

# Reactomics Application 

## Biomarker Reaction

---

## Lung cancer

- MTBLS28 1005 human urine samples

- PMD 2.02 Da show differences among control and diseases

```{r echo=FALSE,out.width="500px"}
knitr::include_graphics('https://yufree.github.io/presentation/figure/mtbls28.png')
```

---

## How

```{r echo=FALSE,out.width='50%', fig.cap='Paper method v.s. Practical method in Metabolomics'}
knitr::include_graphics('https://yufree.github.io/presentation/figure/owl.png')
```


---

## Software

### [pmd package](http://yufree.github.io/pmd/)
  
  - GlobalStd algorithm
  - pmd-reaction database based on KEGG/HMDB
  - quantitative pmd analysis
  - pmd network anaysis
  - pmd MS/MS annotation algorithm
  - GUI application based on shiny

### [rmwf package](https://github.com/yufree/rmwf)

  - NIST 1950 data 
  - Rmarkdown template
  - From raw data to report
  - part of `xcmsrocker` docker image

---

## Take-home message

### Reaction is the "base pair" for small molecules

### Reactomics means the omics study at reaction or PMD level

### Real data always enriched in few PMDs or reactions

### Reaction level changes could be quantitative and predictable

???
Here is the take-home message for reactomics: reaction is the "base pairs" for small molecular, Reactomics means the omics study at reaction or PMD level, real data always enriched in few PMDs or reactions, reaction level changes could be quantitative and predictable.

---

class: inverse, center, middle

# Thanks 

## Q&A

## miao.yu@mssm.edu

```{r cache=FALSE}
knitr::knit_exit() 
```
