<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>Reactomics: Using mass spectrometry as a chemical reaction detector</title>
    <meta charset="utf-8" />
    <meta name="author" content="Miao Yu, Lauren Petrick" />
    <meta name="date" content="2020-05-14" />
    <link href="libs/remark-css/default.css" rel="stylesheet" />
    <link href="libs/remark-css/middlebury.css" rel="stylesheet" />
    <link rel="stylesheet" href="my-theme.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# Reactomics: Using mass spectrometry as a chemical reaction detector
## ID:304104
### Miao Yu, Lauren Petrick
### 2020-05-14

---




## Workflow for metabolomics

- Collect samples

- Collect peaks from mass spectrometry

- Annotate peaks to get the compound name

- Build the links between compounds by pathway/network analysis

&gt; Sample -&gt; Peaks -&gt; Compounds -&gt; Relationship among compounds

- Problems

    - Time consuming - too many peaks ~20k
    - Standards coverage - unknown unknown
    
---

## Idea

&gt; Sample -&gt; Peaks -&gt; ~~Compounds~~ -&gt; Relationship among compounds

- Mass spectrum could directly measure relationsship(reactions)

&lt;img src="https://yufree.github.io/presentation/figure/srda.png" width="100%" style="display: block; margin: auto;" /&gt;

???
- Annotation is not really necessary for certain scientific problem
- Relationship among compounds or reaction matters

---
## Why Reactions?

&lt;img src="https://yufree.github.io/presentation/figure/database.png" style="display: block; margin: auto;" /&gt;

- Unit: Gene(5) &lt; Protein(20+2) &lt; Metabolite(100K) &lt; Compound(100M)

- Combination: Gene(20,000-25,000) &lt; Protein(20,000-25,000) &lt; Compound(???) 

- Small molecular **combination** is chemical reaction or paired mass distance

---

## Why PMD?

- [Nuclear Binding Energy](https://en.wikipedia.org/wiki/Nuclear_binding_energy)

`$$\Delta m = Zm_{H} + Nm_{n} - M$$`
- The missing mass was converted into energy ( `\(E=mc^2\)` ) and emitted when the atom made

- Atoms -&gt; Compounds -&gt; Mass distances between compounds

- **Paired Mass Distances(PMD)** is unique

- **High resolution** mass spectrometry WINs

???
- Mass defects could be transferred from atom to paired mass distance
- HRMS could measure PMDs for qualitative analysis

---

## Sources of PMDs in the real data

### Where is PMD?

.pull-left[- Isotopologues 
  
  - `\([M]^+\)` `\([M+1]^+\)`
  - 1.006 Da

- in source reaction 

  - `\([M+H]^+\)`  `\([M+Na]^+\)`
  - 21.982 Da
]

.pull-right[
- Homologous series

  - Lipid `\(-[CH_2]-\)`
  - 14.016 Da

- Xenobiotic metabolism

  - Phase I hydrolation
  - 15.995 Da
]  


---

## Quantitative and Qualitative analysis for Reaction

### KEGG reaction database


|   PMD   | Freq | ExampleReactionClass | ExampleEnzyme |
|:-------:|:----:|:--------------------:|:-------------:|
|  2.016  | 1732 |       RC00095        |   1.3.1.84    |
| 15.995  | 1169 |       RC00046        |   1.3.99.18   |
| 79.966  | 729  |       RC00002        |    3.6.1.3    |
| 14.016  | 594  |       RC00060        |    1.5.3.1    |
|    0    | 532  |       RC00302        |    5.1.1.3    |
| 18.011  | 359  |       RC00680        |    3.5.2.5    |
| 162.053 | 365  |       RC00049        |   3.2.1.23    |
| 159.933 | 243  |       RC02056        |   4.2.3.42    |
|  1.032  | 262  |       RC00006        |    1.4.1.2    |
| 42.011  | 237  |       RC00004        |   2.3.1.54    |

---

## Quantitative and Qualitative analysis for Reaction

### Static v.s. dynamic 

- Static mass pairs: paired intensity ratio is stable across samples
- Dynamic mass pairs: paired intensity ratio is stable across samples
- For example, [A,B], [C,D] and [E,F] are involved in the same PMD:


|  A   |  B  | Ins ratio |  C  | D  | Ins ratio |  E  |  F  | Ins ratio |
|:----:|:---:|:---------:|:---:|:--:|:---------:|:---:|:---:|:---------:|
| 100  | 50  |    2:1    | 100 | 50 |    2:1    | 30  | 40  |    3:4    |
| 1000 | 500 |    2:1    | 10  | 95 |   2:19    | 120 | 160 |    3:4    |

- [A,B] and [E,F] could be used for Quantitative analysis for certain PMD, rsd cutoff 30%
- [C,D] could be used to check dynamics of specific reaction

???
- Response factor is the slope of calibration curve for certain compound
- Total intensity of all pairs with the same PMD
- Count once for ions involved in multiple reactions

---
class: inverse, center, middle

# Reactomics Application 

## Metabolites Discovery

---

## Metabolites of exogenous compound

  - Environmental pollution metabolites
  - Drug metabolites

### Xenobiotic metabolism

  - Phase I
    - Oxidation (R-H ⇒ R-OH, pmd 15.995 Da)
    - Reduction (R-C=O ⇒ R-C-OH, pmd 2.016 Da)
    
  - Phase II
    - Methylation (R-OH ⇒ R-O-C,pmd 14.016 Da)
    - Sulfation (R-OH ⇒ R-SO4, pmd 46.976 Da)
    - Acetylation (R-OH ⇒ R-O-COCH3, pmd 42.011 Da)
    - Glucuronidation (R-NH2 ⇒ R-NH-C6H9O7, pmd 192.027 Da)
    - Glycosylation (R-OH ⇒ R-O-C6H11O5, pmd 162.053 Da)

---

## Metabolites of TBBPA in Pumpkin

- Mass defect analysis to screen Brominated Compounds 

- Confirmation by synthesized standards

&lt;img src="https://yufree.github.io/presentation/figure/TBBPAmet.jpeg" width="600px" style="display: block; margin: auto;" /&gt;

.half[
Hou, X., Yu, M., Liu, A., Wang, X., Li, Y., Liu, J., … Jiang, G. (2019). Glycosylation of Tetrabromobisphenol A in Pumpkin. Environmental Science &amp; Technology, 53(15), 8805–8812. doi:10.1021/acs.est.9b02122
]

---

## Metabolites of TBBPA in Pumpkin

- TBBPA Metabolites PMD network

&lt;img src="https://yufree.github.io/presentation/figure/TBBPA.png" width="500px" style="display: block; margin: auto;" /&gt;


.half[
Hou, X., Yu, M., Liu, A., Wang, X., Li, Y., Liu, J., … Jiang, G. (2019). Glycosylation of Tetrabromobisphenol A in Pumpkin. Environmental Science &amp; Technology, 53(15), 8805–8812. doi:10.1021/acs.est.9b02122
]

---
class: inverse, center, middle

# Reactomics Application 

## Endogenous or Exogenous

---

## Endogenous vs Exogenous

- T3DB Endogenous (255) vs Exogenous (705)

- Use top 10 high frequency PMDs

&lt;img src="https://raw.githubusercontent.com/yufree/presentation/gh-pages/figure/t3db3.png" width="450px" style="display: block; margin: auto;" /&gt;

---

## KEGG reaction network

- Metabolites of four compounds

- Average network distance: Glucose(9.74), 5-Cholestene(6.55), Caffeine(3.31), Bromophenol(1.8), TBBPA(3.88 for pumpkin)

&lt;img src="https://raw.githubusercontent.com/yufree/presentation/gh-pages/figure/kegg.png" width="450px" style="display: block; margin: auto;" /&gt;

---

class: inverse, center, middle

# Reactomics Application 

## Biomarker Reaction

---

## Lung cancer

- MTBLS28 1005 human urine samples

- PMD 2.02 Da show differences among control and diseases

&lt;img src="https://yufree.github.io/presentation/figure/mtbls28.png" width="500px" style="display: block; margin: auto;" /&gt;

---

## How

&lt;div class="figure" style="text-align: center"&gt;
&lt;img src="https://yufree.github.io/presentation/figure/owl.png" alt="Paper method v.s. Practical method in Metabolomics" width="72%" /&gt;
&lt;p class="caption"&gt;Paper method v.s. Practical method in Metabolomics&lt;/p&gt;
&lt;/div&gt;

---

## Software

### [pmd package](http://yufree.github.io/pmd/)
  
  - GlobalStd algorithm
  - pmd-reaction database based on KEGG/HMDB
  - quantitative pmd analysis
  - pmd network anaysis
  - pmd MS/MS annotation algorithm
  - GUI application based on shiny

### [rmwf package](https://github.com/yufree/rmwf)

  - NIST 1950 data 
  - Rmarkdown template
  - From raw data to report
  - part of `xcmsrocker` docker image
  
### POSTER 304118

---

class: inverse, center, middle

# Thanks 

## Q&amp;A

## miao.yu@mssm.edu
    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true,
"countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
